mlflow/models/docker_utils.py

```diff
@@ -43,42 +43,77 @@
+# Build an image that can serve mlflow models.
+FROM ubuntu:18.04
+
+RUN apt-get -y update && apt-get install -y --no-install-recommends \
+         curl \
+         nginx \
+    && rm -rf /var/lib/apt/lists/*
+
+# Download and setup miniconda
+RUN curl https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh >> miniconda.sh
+RUN bash ./miniconda.sh -b -p /miniconda; rm ./miniconda.sh;
+ENV PATH="/miniconda/bin:$PATH"
+ENV GUNICORN_CMD_ARGS="--timeout 60 -k gevent"
+# Set up the program in the image
+WORKDIR /opt/mlflow
+
+{install_mlflow}
+
+{custom_setup_steps}
+{entrypoint}
+"""
 
-def _get_mlflow_install_step(dockerfile_context_dir, mlflow_home):
+
+def _get_mlflow_install_step(dockerfile_context_dir, mlflow_home, python_only):
     """
     Get docker build commands for installing MLflow given a Docker context dir and optional source
     directory
     """
     if mlflow_home:
         mlflow_dir = _copy_project(
             src_path=mlflow_home, dst_path=dockerfile_context_dir)
-        return (
-            "COPY {mlflow_dir} /opt/mlflow\n"
-            "RUN pip install /opt/mlflow\n"
-            "RUN cd /opt/mlflow/mlflow/java/scoring && "
-            "mvn --batch-mode package -DskipTests && "
-            "mkdir -p /opt/java/jars && "
-            "mv /opt/mlflow/mlflow/java/scoring/target/"
-            "mlflow-scoring-*-with-dependencies.jar /opt/java/jars\n"
-        ).format(mlflow_dir=mlflow_dir)
+        if python_only:
+            return (
+                "COPY {mlflow_dir} /opt/mlflow\n"
+                "RUN pip install /opt/mlflow\n"
+                ).format(mlflow_dir=mlflow_dir)
+        else:
+            return (
+                "COPY {mlflow_dir} /opt/mlflow\n"
+                "RUN pip install /opt/mlflow\n"
+                "RUN cd /opt/mlflow/mlflow/java/scoring && "
+                "mvn --batch-mode package -DskipTests && "
+                "mkdir -p /opt/java/jars && "
+                "mv /opt/mlflow/mlflow/java/scoring/target/"
+                "mlflow-scoring-*-with-dependencies.jar /opt/java/jars\n"
+                ).format(mlflow_dir=mlflow_dir)
     else:
-        return (
-            "RUN pip install mlflow=={version}\n"
-            "RUN mvn "
-            " --batch-mode dependency:copy"
-            " -Dartifact=org.mlflow:mlflow-scoring:{version}:pom"
-            " -DoutputDirectory=/opt/java\n"
-            "RUN mvn "
-            " --batch-mode dependency:copy"
-            " -Dartifact=org.mlflow:mlflow-scoring:{version}:jar"
-            " -DoutputDirectory=/opt/java/jars\n"
-            "RUN cp /opt/java/mlflow-scoring-{version}.pom /opt/java/pom.xml\n"
-            "RUN cd /opt/java && mvn "
-            "--batch-mode dependency:copy-dependencies -DoutputDirectory=/opt/java/jars\n"
-        ).format(version=mlflow.version.VERSION)
-
-
-def _build_image(image_name, entrypoint, mlflow_home=None, custom_setup_steps_hook=None):
+        if python_only:
```

Comment:
Similar to above, build the string, add to it `if not python_only`, and then format and return.